### DISCUSSION TITLE
How to restart an ionic capacitor app?

### DISCUSSION THREAD

**Discussion by distante** [ionic-team/capacitor#3922] [Q&A]
I am looking for ways to trigger a restart of my app programmatically. That is start everything again from the splash screen and load all plugins again.

Please notice, I am not talking about refreshing the WebView but a full app restart. Doing a WebView refresh will duplicate event listeners registered on Native plugins and I want to avoid this.

**Comment by mlynch** [comment]
This is not something you can or should do for an app, Capacitor or otherwise.

One thing you could do if you need to restart something is bind native listeners on a lifecycle enter (or useEffect in React for example), and unbind on lifecycle exit (or the return function in useEffect in React, for example). Then you won't have the issue of duplicated listeners.

**Reply by distante** [reply]
There are some global listeners like `AdMob` events or media events that should be registered in every component of my app (angular).

My specific use case happens when the WebView web audio context gets broken when a Bluetooth device is connected, it is a known bug in Chrome. 

Until (and if) it gets fixed I want to just reload the app when a Bluetooth device is connected. It is not pretty, but change the complete implementation for a bug in the android Webview to recreate audio contexts and chains of connections is really a lot of work. 

I saw that there are some native libs that do this on Android. I suppose I will try to write a capacitor plugin for this.

**Reply by mlynch** [reply]
I suppose you could access the web view from a plugin and then force it to reload

**Reply by distante** [reply]
But reloading the webview thru the native code will not reset the listeners registered on the Native plugins. Correct? 🤔

**Reply by mlynch** [reply]
Correct, you would need to unregister those before doing the reload
